ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/strategy/transferstrategy/AbstractTransferStrategy.java

```diff
@@ -117,20 +127,21 @@ public void transfer() throws IOException {
     }
 
     private void transferAllJobBlobs() throws IOException {
+        final Predicate transferPredicate = () -> ! canceled;
         final AtomicInteger numBlobsRemaining = new AtomicInteger(jobState.numBlobsInJob());
 
-        while (!Thread.currentThread().isInterrupted() && numBlobsRemaining.get() > 0) {
+        while ( ! Thread.currentThread().isInterrupted() && transferPredicate.test() && numBlobsRemaining.get() > 0) {
```

Comment:
what is `transferPredicate.test()`?

Reply:
It's a lambda that evaluates `! canceled`  It's defined on line 130.